### PR TITLE
Implement Oauth in PGObserver

### PR DESCRIPTION
**Pull request by evakbe**

### CODE REVIEW

File: frontend/src/web.py [hunk 1]
```diff
@@ -120,6 +120,12 @@ def main():
     root.hosts = hostsfrontend.HostsFrontend()
     root.api = api.Root(root)   # JSON api exposure, enabling integration with other monitoring tools
 
+    if settings.get('oauth2',{}).get('switch'):
```
Comment (a1exsh) [inline frontend/src/web.py]: Shouldn't it be like: if ....get('switch', 'off') == 'on': ?

File: frontend/src/zalandoauth.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        self.client_id = oauth2['client_id']
+        self.client_secret = oauth2['client_secret']
+        self.access_token_url = oauth2['access_token_url']
+        self.authorize_url = oauth2['authorize_url']
+        self.redirect_url = oauth2['redirect_url']
+
+    def logout(self):
+        cherrypy.sessions.expire()
+        raise cherrypy.HTTPRedirect('/')
+    logout.expose = True
```
Comment (a1exsh) [inline frontend/src/zalandoauth.py]: use @cherrypy.expose decorator instead?

**Comment by hjacobs** [comment]
Would be nice if you guys could start using "flake8" to check your Python code style and formatting :smirk:

:point_right: https://github.com/PyCQA/flake8

File: frontend/src/zalandoauth.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                cherrypy.session[FLAG_access_token] = access_token
+                response.close()
+                # redirect to endpoint where user attempted to access
+                if target_url:
+                    raise cherrypy.HTTPRedirect(target_url)
+                else:
+                    raise cherrypy.HTTPRedirect('/')
+            else:
+                response.close()
+                raise Exception(response.json())
```
Comment (a1exsh) [inline frontend/src/zalandoauth.py]: Are we not exposing any sensitive bits by doing that?

File: frontend/src/web.py [hunk 4]
```diff
@@ -120,6 +120,12 @@ def main():
...
+        print 'switching on oauth2 ...'
+        import zalandoauth
+        root.zalandoauth = zalandoauth.ZalandOauth(settings)
```
Comment (alexeyklyukin) [inline frontend/src/web.py]: no need to pass all settings, if you only need settings['oauth2'], that'd also improve isolation of modules.

File: frontend/pgobserver.yaml [hunk 5]
```diff
@@ -12,6 +12,14 @@ database:
 frontend:
   port: 8080
 
+oauth2:
+  #switch: on    #comment this out if oauth is not desired
```
Comment (alexeyklyukin) [inline frontend/pgobserver.yaml]: Naming nitpick: 'enable' is more self-explanatory then 'switch'.

File: frontend/src/web.py (hunk 1 above)
Comment (evakbe) [inline frontend/src/web.py]: I thought so too; I discovered yaml-load interprets that to python-true.
Just to stay out of translation issues I decided for exists or not (should cover boolean interpretations too).

File: frontend/pgobserver.yaml (hunk 5 above)
Comment (evakbe) [inline frontend/pgobserver.yaml]: I'm actually going to remove this and use the other existing params

File: frontend/src/zalandoauth.py (hunk 3 above)
Comment (evakbe) [inline frontend/src/zalandoauth.py]: potentially yes, but for the cases I checked the endpoints don't spill any secrets.

File: frontend/src/zalandoauth.py (hunk 2 above)
Comment (evakbe) [inline frontend/src/zalandoauth.py]: ready for commit

File: frontend/src/web.py (hunk 4 above)
Comment (evakbe) [inline frontend/src/web.py]: ready for commit

File: frontend/src/web.py [hunk 6]
```diff
@@ -72,30 +69,16 @@ def main():
...
+    conf = {
+        'global': {'server.socket_host': '0.0.0.0', 'server.socket_port': args.port or settings.get('frontend',
+                   {}).get('port') or 8080},
+        '/': {'tools.staticdir.root': current_dir},
+        '/static': {'tools.staticdir.dir': 'static', 'tools.staticdir.on': True},
+        '/manifest.info': {'tools.staticfile.on': True, 'tools.staticfile.filename': os.path.join(current_dir, '..',
+                           'MANIFEST.MF'), 'tools.auth_basic.on': False},
+    }
+
+    tplE.setup(settings)  # setup of global variables and host data for usage in views
```
Comment (a1exsh) [inline frontend/src/web.py]: Hm, I don't think this re-formatting can be justified.  The original might be imperfect, but the new version doesn't look all that better.  Not to mention it is totally unrelated to the actual change in this branch.

**Pull request merged**